README.md

```diff
@@ -18,3 +18,30 @@ require('tree-sitter-typescript').tsx; // TSX grammar
 References
 
 * [TypeScript Language Spec](https://github.com/microsoft/TypeScript/blob/main/doc/spec-ARCHIVED.md)
+
+## Development
+
+This repository contains the grammar definition in `common/define-grammar.js`, as well as query files in the `queries` directory.
+The query files specify highlighting in `queries/highlights.scm`, and global and local names in `queries/tags.scm` and `queries/locals.scm`, respectively.
+Name binding rules are defined using [stack-graphs][] in `queries/stack-graphs.tsg`.
+Tests for the grammar and the stack graph rules can be found in `common/corpus` and `typescript/test/stack-graphs`, respectively.
+
+Build the grammar as follows:
+
+```
+$ npm run build
+```
+
+Run the tests as follows (this requires the [cargo][] command to be installed):
+
+```
+$ npm run test
+```
+
+Contributors are expected to keep query files and stack graph up to date with the grammar.
+Depending on the changes, updating the stack graph rules may not always be possible for contributors, as it requires knowledge of stack graphs.
+This should not prevent development on the grammar and in such cases the @tree-sitter/semantic-code team should be pinged for support.
+They will either help the contributor make the necessary changes, or approve a temporary disabling of stach graph tests in CI, so that the changes can be carried out at a later time.
```

Comment:
We want the stack graph rules to live in the grammar repo and have tests run in CI. However, because stack graphs require knowledge beyond what is necessary for grammar development, we don't want this to be blocking grammar contributions either. This is a proposed way of dealing with that tension.